share/admin/readonly.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+from django.contrib import admin
+from django.core.exceptions import PermissionDenied
+
+
+class SetOfEverything(list):
+    def __contains__(self, other):
+        return True
+
+
+## adopted from http://mike.hostetlerhome.com/blog/2012/11/13/add-a-read-only-role-to-django-admin/
+class ReadOnlyAdmin(admin.ModelAdmin):
+
+    def has_add_permission(self, request, obj=None):
+        return self.__user_is_readonly(request)
```

Comment:
I think so. It's pretty hacky, you have to give a user some kind of permission to view objects. In the rest of the code it disallows all action :/